web-components/src/components/form/CreditTransferForm.tsx

```diff
@@ -0,0 +1,315 @@
+      }
+    }
+
+    return errors;
+  };
+
+  const submitHandler = async (values: FormValues): Promise<any> => {
+    console.log('*** submitHandler', values);
+    // TODO holder, amount string
+  };
+
+  const countryHandler = (countryCode: string): any => {
+    console.log('*** countryHandler', countryCode);
+  };
+
+  const stateCountryHandler = (stateCountry: string): any => {
+    console.log('*** stateCountryHandler', stateCountry);
+  };
+
+  return (
+    <Formik
+      initialValues={initialValues}
+      validate={validateHandler}
+      onSubmit={submitHandler}
+    >
+      {({ values, submitForm, isSubmitting, isValid, submitCount, status }) => (
+        <Form>
+          <Field
+            name="sender"
+            type="text"
+            label="Sender"
+            component={TextField}
+            className={styles.senderField}
+            disabled
+          />
+          <Field
+            name="recipient"
+            type="text"
+            label="Recipient"
+            component={TextField}
+            className={styles.textField}
+          />
+          <Field
+            name="amount"
+            type="number"
+            component={TextField}
+            className={styles.textField}
+            label={
+              <AmountLabel
+                label={'Amount to retire'}
+                tradableAmount={tradableAmount}
+                batchDenom={batchDenom}
+              />
+            }
+          />
+
+          <Field
+            component={CheckboxLabel}
+            type="checkbox"
+            name="retire"
+            className={styles.checkboxLabel}
+            label={<Description>Retire credits upon transfer</Description>}
+          />
+
+          {values.retire && (
+            <>
+              <Field
+                name="amountRetire"
+                type="number"
+                component={TextField}
+                className={styles.textField}
+                label={
+                  <AmountLabel
+                    label={'Amount to retire'}
+                    tradableAmount={tradableAmount}
+                    batchDenom={batchDenom}
+                  />
+                }
+              />
+              <Title className={styles.groupTitle} variant="h5">
+                Location of retirement
+              </Title>
+              <Description className={styles.description}>
+                Please enter a location for the retirement of these credits.
+                This prevents double counting of credits in different locations.
+              </Description>
+              <Grid
+                container
+                alignItems="center"
+                className={styles.stateCountryGrid}
+              >
+                <Grid
+                  item
+                  xs={12}
+                  sm={6}
+                  className={styles.stateCountryTextField}
+                >
+                  <LocationStateCountryField
```

Comment:
why not using `Field` component from formik as for the other fields?

Reply:
I simply extracted this component that was already created in another form to bring it into the component library and make it more reusable. Inside it uses a Formik Field component